src/taipy/core/data/data_node.py

```diff
@@ -286,8 +286,8 @@ def __get_last_modified_datetime(self) -> Optional[datetime]:
 
     @classmethod
     @abstractmethod
-    def storage_type(cls) -> str:
-        raise NotImplementedError()
+    def storage_type(cls) -> Optional[str]:
+        return None
```

Comment:
I don't understand this change. 

This is an abstract method that must be overridden by all implementation classes. So if this code is executed, it must raise an error. 

Am I missing something?

Reply:
I agree. However, this code is called by https://github.com/Avaiga/taipy-core/blob/5173b8e48fd29c6d0a360c8f2ff39da51f97791a/src/taipy/core/data/data_node.py#L521. Raising an exception from this method would break a bit of everything. The code assume the storage type can be `None` so returning None here looks like the only option for now...

Reply:
That also explains why the type for storage type is now `Optional[str]`. 
Or maybe we should ask the storage type only for concrete DataNode subclasses?

Reply:
Since the `DataNode` class should not be used by the end user, and we want all children of `DataNode` to have `storage_type`, I don't think we should change this.

Do we have any other method to avoid mypy error message here? Like a `# type: ignore` or something.

Reply:
Yes, I agree @trgiangdo... Changing the implementation of `storage_type()` causes cascading other changes that goes beyond the scope of this PR. 
If you all agree, we shall keep `return NotImplemented` for now in this method, and this would be addressed in another dedicated PR.
Wdyt?

Reply:
yes, it should be sufficient for now. Thank you